homeassistant/components/recorder/migration.py

```diff
@@ -660,7 +660,7 @@ def _inspect_schema_version(engine, session):
     version 1 are present to make the determination. Eventually this logic
     can be removed and we can assume a new db is being created.
     """
-    inspector = sqlalchemy.inspect(engine)
+    inspector = sqlalchemy.inspect(session.connection())
```

Comment:
If we create an inspector directly on the engine, the inspector's connection will mess with the session when it's shutting down. This may be a bug in SQLite.